netplan/cli/commands/apply.py

```diff
@@ -252,7 +252,13 @@ def command_apply(self, run_generate=True, sync=False, exit_on_error=True, state
                            if not f.endswith('/' + OVS_CLEANUP_SERVICE)]
             # Run 'systemctl start' command synchronously, to avoid race conditions
             # with 'oneshot' systemd service units, e.g. netplan-ovs-*.service.
-            utils.networkctl_reconfigure(utils.networkd_interfaces())
+            try:
+                utils.networkctl_reload()
+                utils.networkctl_reconfigure(utils.networkd_interfaces())
+            except subprocess.CalledProcessError:
+                # (re-)start systemd-networkd if it is not running, yet
+                logging.warning('Falling back to a hard restart of systemd-networkd.service')
+                utils.systemctl('restart', ['systemd-networkd.service'], sync=True)
```

Comment:
Generally, I don't like try/except blocks, and especially when they wrap two commands without distinction if networkd is not running (hence networctl fails to connect to networkd) or if there is something broken with the config (and thus reconfigure of a device fails, but networkd is running and gets pointlessly restarted).

Can you check if systemd-networkd.service is running? If yes, execute networkctl reload & reconfigure. Otherwise check if networkd is masked and fail. If it is not masked, restart it.

Reply:
> Can you check if systemd-networkd.service is running? If yes, execute networkctl reload & reconfigure. Otherwise restart networkd.

That is actually how I had it implemented initially, but it does not catch cases where systemd-networkd.service is running but "stuck" (for some unkown reason), as reported in the original bug report at LP#1962095 (comment 4). Sure, networkd being stuck is not really a netplan problem... but using the try/except block we can hit two birds with one stone.